tsdb/db.go

```diff
@@ -835,8 +835,12 @@ func (db *DB) compactHead(head *RangeHead) (err error) {
 	runtime.GC()
 
 	if err := db.reload(); err != nil {
-		if err := os.RemoveAll(filepath.Join(db.dir, uid.String())); err != nil {
-			return errors.Wrapf(err, "delete persisted head block after failed db reload:%s", uid)
+		var merr tsdb_errors.MultiError
```

Comment:
`merr` is only requred inside `os.RemoveAll` error, so we can declare inside the next `if`.